src/utils/metamaskUtils.ts

```diff
@@ -0,0 +1,18 @@
+const registerToken = async (earningTokenAddress: string, tokenName: string, tokenDecimals: number, image: string) => {
```

Comment:
Try to be consistent in your names choices. `earningTokenAddress` -> `tokenAddress`